de-lib/src/main/java/org/iplantc/de/apps/client/views/navigation/AppNavigationViewImpl.java

```diff
@@ -0,0 +1,120 @@
+package org.iplantc.de.apps.client.views.navigation;
+
+import org.iplantc.de.apps.client.AppNavigationView;
+
+import com.google.common.collect.Lists;
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.event.logical.shared.SelectionEvent;
+import com.google.gwt.event.logical.shared.SelectionHandler;
+import com.google.gwt.event.shared.HandlerRegistration;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiFactory;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+
+import com.sencha.gxt.data.shared.StringLabelProvider;
+import com.sencha.gxt.widget.core.client.Composite;
+import com.sencha.gxt.widget.core.client.container.CardLayoutContainer;
+import com.sencha.gxt.widget.core.client.container.VerticalLayoutContainer;
+import com.sencha.gxt.widget.core.client.form.SimpleComboBox;
+
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * @author aramsey
+ */
+public class AppNavigationViewImpl extends Composite implements AppNavigationView {
+
+    interface AppNavigationViewImplUiBinder extends UiBinder<Widget, AppNavigationViewImpl> {}
+    private static final AppNavigationViewImplUiBinder uiBinder = GWT.create(AppNavigationViewImplUiBinder.class);
+
+    @UiField VerticalLayoutContainer vlc;
+    @UiField SimpleComboBox<String> combo;
+    @UiField CardLayoutContainer container;
+    private HashMap<Widget, String> selectionMap;
+
+    @Inject
+    public AppNavigationViewImpl() {
+        initWidget(uiBinder.createAndBindUi(this));
+        selectionMap = new HashMap<>();
+    }
+
+    @UiFactory
+    SimpleComboBox<String> createComboBox() {
+        SimpleComboBox<String> combo = new SimpleComboBox<>(new StringLabelProvider<>());
+        combo.addSelectionHandler(event -> {
+            String selectedName = event.getSelectedItem();
+            int index = combo.getStore().indexOf(selectedName);
+            container.setActiveWidget(container.getWidget(index));
+            SelectionEvent.fire(this, container.getActiveWidget());
+        });
+        return combo;
+    }
+
+    @Override
+    public void mask() {
+        vlc.mask();
+    }
+
+    @Override
+    public void unmask() {
+        vlc.unmask();
+    }
+
+    @Override
+    public void add(Widget widget, String name) {
+        insert(widget, container.getWidgetCount(), name);
+    }
+
+    @Override
+    public void insert(Widget widget, int index, String name) {
+        if (!selectionMap.containsKey(widget)) {
```

Comment:
I was wondering why not use a regular `ComboBox` over the `SimpleComboBox` + this `selectionMap`, but it does seem this widget lookup here is much simpler than trying to find the widget in a `ComboBox` list store 🤔

Reply:
Well for one, I totally forgot about the regular `ComboBox` lol
But if I do remember correctly, I think `ComboBox` was better suited for a list of items of the same type which all had the same method/property that could be used to extract the display name. This is a combo of `Tree`s but they're not all of the same type (OntologyHierarchy, AppCategory, Group), so I'm not sure that would work.  I could be wrong though, it's been a while since I've used one.